File: solutions/python/chessgame/player.py
class Player:
    def __init__(self, color):
        self.color = color

    def make_move(self, board, move):
        piece = move.piece
        dest_row = move.dest_row
        dest_col = move.dest_col

        if board.is_valid_move(piece, dest_row, dest_col):
            source_row = piece.row
            source_col = piece.col
            board.set_piece(source_row, source_col, None)
            board.set_piece(dest_row, dest_col, piece)
            piece.row = dest_row
            piece.col = dest_col
        else:
            raise ValueError("Invalid move!")

File: solutions/csharp/chessgame/Player.cs
namespace ChessGame
{
    public class Player
    {
        public Color Color { get; }

        public Player(Color color)
        {
            Color = color;
        }

        public void MakeMove(Board board, Move move)
        {
            Piece piece = move.Piece;
            int destRow = move.DestRow;
            int destCol = move.DestCol;

            if (board.IsValidMove(piece, destRow, destCol))
            {
                int sourceRow = piece.Row;
                int sourceCol = piece.Col;
                board.SetPiece(sourceRow, sourceCol, null);
                board.SetPiece(destRow, destCol, piece);
                piece.Row = destRow;
                piece.Col = destCol;
            }
            else
            {
                throw new InvalidMoveException("Invalid move!");
            }
        }
    }
}